src/xma/src/xmaapi/xmaxclbin.cpp

```diff
@@ -35,27 +36,32 @@ static int get_xclbin_iplayout(char *buffer, XmaXclbinInfo *xclbin_info);
 static int get_xclbin_mem_topology(char *buffer, XmaXclbinInfo *xclbin_info);
 static int get_xclbin_connectivity(char *buffer, XmaXclbinInfo *xclbin_info);
 
-char *xma_xclbin_file_open(const char *xclbin_name)
+std::vector<char> xma_xclbin_file_open(const char *xclbin_name)
 {
     xma_logmsg(XMA_INFO_LOG, XMAAPI_MOD, "Loading %s\n", xclbin_name);
 
-    std::ifstream file(xclbin_name, std::ios::binary | std::ios::ate);
-    std::streamsize size = file.tellg();
-    file.seekg(0, std::ios::beg);
+    std::ifstream infile(xclbin_name, std::ios::binary | std::ios::ate);
+    std::streamsize xclbin_size = infile.tellg();
+    infile.seekg(0, std::ios::beg);
 
-    char *buffer = (char*)malloc(size);
-    if (buffer == NULL) {
+    std::vector<char> xclbin_buffer;
```

Comment:
std::vector<char> xclbin_buffer(size), no need for separate reserve

Reply:
keeping it to capture alloc exception error in log.

Reply:
std::vector xclbin_buffer(size) will throw std::bad_alloc.  You are not achieving anything with separate call to reserve.

Reply:
below that catching bad_alloc exception with reserve for the log msg